fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/ProgressiveLoanScheduleGenerator.java

```diff
@@ -280,6 +280,9 @@ public OutstandingAmountsDTO calculatePrepaymentAmount(MonetaryCurrency currency
                 .principal(result.getOutstandingBalance()) //
                 .interest(result.getPayableInterest());
 
+        installments.stream().filter(installment -> installment.getDueDate().isBefore(onDate) || installment.getDueDate().isEqual(onDate))
```

Comment:
I am afraid the `|| installment.getDueDate().isEqual(onDate)` is still incorrect as if the `onDate` is on the same day as the installment due date, that means that is the "actual" installment and we already calculated the payable interest for the actual installment.